spyder/plugins/__init__.py

```diff
@@ -538,12 +538,30 @@ class SpyderPluginWidget(QWidget, SpyderPluginMixin):
 
     if PYQT5:
         def __init__(self, parent, **kwds):
+            check_compatibility, message = self.check_compatibility()
             super(SpyderPluginWidget, self).__init__(parent, **kwds)
+            if not check_compatibility:
+                messageBox = QMessageBox(self)
+                messageBox.setWindowModality(Qt.NonModal)
+                messageBox.setAttribute(Qt.WA_DeleteOnClose)
+                messageBox.setWindowTitle('Compatibility Check')
+                messageBox.setText(message)
+                messageBox.setStandardButtons(QMessageBox.Ok)
+                messageBox.show()
     else:
         def __init__(self, parent):
+            check_compatibility, message = self.check_compatibility()
             QWidget.__init__(self, parent)
             SpyderPluginMixin.__init__(self, parent)
-        
+            if not check_compatibility:
+                messageBox = QMessageBox(self)
+                messageBox.setWindowModality(Qt.NonModal)
+                messageBox.setAttribute(Qt.WA_DeleteOnClose)
+                messageBox.setWindowTitle('Compatibility Check')
+                messageBox.setText(message)
+                messageBox.setStandardButtons(QMessageBox.Ok)
+                messageBox.show()

```

Comment:
Let's put all this  in another method called show_compatibility_message, to not repeat twice in different places.